octopusdeploy3-single-vm-windows/azuredeploy.json

```diff
@@ -0,0 +1,355 @@
+        "North Europe",
+        "South Central US",
+        "Southeast Asia",
+        "West Europe",
+        "West US"
+      ],
+      "metadata": {
+        "Description": "Location of the Octopus Deploy server."
+      }
+    },
+    "vmStorageAccountName": {
+      "type": "string",      
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account that will hold the Octopus Deploy Virtual Machine Disks."
+      }
+    },
+    "vmAdminUsername": {
+      "type": "string",
+      "defaultValue": "octoadmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "networkDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name used to access the Octopus Deploy server via HTTP or RDP."
+      }
+    },
+    "sqlServerName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the SQL DB Server that will hold the Octopus Deploy data."
+      }
+    },
+    "sqlAdminUsername": {
+      "type": "string",
+      "defaultValue": "sqladmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "sqlAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "licenseFullName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Full Name."
+      }
+    },
+    "licenseOrganisationName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Organisation Name."
+      }
+    },
+    "licenseEmailAddress": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Email Addres."
+      }
+    },
+    "octopusAdminUsername" : {
+      "type": "string",
+      "defaultValue": "admin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy web application."
+      }
+    },
+    "octopusAdminPassword" : {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy web application."
+      }
+    }
+  },
+  "variables": {
+    "location": "[parameters('location')]",
+    "vmImagePublisher": "MicrosoftWindowsServer",
+    "vmImageOffer": "WindowsServer",
+    "vmOSDiskName": "osdiskforwindowssimple",
+    "vmWindowsOSVersion": "2012-R2-Datacenter",
+    "vmStorageAccountName": "[parameters('vmStorageAccountName')]",
+    "vmStorageAccountType": "Standard_LRS",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "OctopusDeploy",
+    "vmSize": "Standard_D2",
```

Comment:
I initially had this as a parameter, but the user experience of having to understand the **magic strings** when deploying via the Azure Portal felt bad to me. 

It would be great to have access to a lookup function in the ARM Templates that would map the user input (from allowed values list) **Standard D2 (2 cores, 7GB) ** to **Standard_D2** or **Standard A2 (2 cores, 3.5GB)** to **medium**.

In the end I decided to fix the value. This is the suggested spec for a single Octopus VM and this template is aimed at **kicking the tires** on an Octopus 3.0 environment.

I'm happy to make it a limited list but not sure the flexibility (and extra parameter) makes sense here.